components/layout_2020/display_list/mod.rs

```diff
@@ -254,9 +268,15 @@ impl<'a> BuilderForBoxFragment<'a> {
             border_edge_clip_id: OnceCell::new(),
             padding_edge_clip_id: OnceCell::new(),
             content_edge_clip_id: OnceCell::new(),
+            root_fragment_background_style,
         }
     }
 
+    fn background_style(&self) -> &ServoArc<ComputedValues> {
+        self.root_fragment_background_style
+            .unwrap_or(&self.fragment.style)
```

Comment:
If the canvas ends up taking `background-image` from a `<body>` element, what stops that image from being painted a second time when generating display lists for that element?

Reply:
Since background positioning is unchanged, observing this requires alpha transparency to be involved. Test case in case there isn’t already one in WPT: http://software.hixie.ch/utilities/js/live-dom-viewer/saved/7907

Reply:
Yep, it seems like this is still an issue.